src/eh1/pin.rs

```diff
@@ -105,6 +123,10 @@ pub enum TransactionKind {
     Set(State),
     /// Get the pin state
     Get(State),
+    /// Toggle for an expected toggle of the pin
+    Toggle,
+    /// Get the set state of the stateful pin
+    GetSetState(State),
```

Comment:
Should we rename this to `GetState`? Otherwise users could be confused: Will it get or set the state? 😄 

The factory method would also need a rename to `get_set_state`.

Reply:
I kinda feel like there is no good name for this :sweat_smile:. I thought that `GetState(State)` might be confused with `Get(State)` so I opted for this. I'm happy to change it though, I could also do `GetTheSetState` or `GetCurrentSetState` but they're both a bit verbose, maybe `GetCurrentState`? :shrug:

Reply:
I think `GetState` is fine. If you prefer `GetCurrentState`, that would be OK for me too.

Reply:
I went for `GetState` in the end.